rts/Sim/Misc/QuadField.cpp

```diff
@@ -523,17 +522,27 @@ void CQuadField::AddProjectile(CProjectile* p)
 		p->quads = std::move(*qfQuery.quads);
 	} else {
 		int newQuad = WorldPosToQuadFieldIdx(p->pos);
+		if (!p->synced) {
+			spring::VectorInsertUnique(baseQuads[newQuad].particles, p, false);
+			p->quads.clear();
+			p->quads.push_back(newQuad);
+			return;
+		}
+
 		spring::VectorInsertUnique(baseQuads[newQuad].projectiles, p, false);
 		p->quads.clear();
 		p->quads.push_back(newQuad);
```

Comment:
Couldn't this be simplified to something like:

`std::vector<CProjectile *>& vector = p->synced ? baseQuads[newQuad].projectiles : baseQuads[newQuad].particle;
spring::VectorInsertUnique(vector, p, false);`
`p->quads.clear();`
`p->quads.push_back(newQuad);`